arcade/backgrounds.py

```diff
@@ -0,0 +1,408 @@
+        :param angle: The BackgroundTexture angle.
+        :param filters: The OpenGl Texture filters (gl.Nearest by default).
+        :param shader: The shader used for rendering.
+        :param geometry: The geometry used for rendering (a rectangle equal to the size by default).
+        :return: The generated Background.
+        """
+        _context = get_window().ctx
+
+        with Image.open(resolve_resource_path(tex_src)).convert("RGBA") as img:
+            texture = _context.texture(img.size, data=img.transpose(Image.FLIP_TOP_BOTTOM).tobytes(),
+                                       filter=filters)
+            if size is None:
+                size = texture.size
+
+        background_texture = BackgroundTexture(texture, offset, scale, angle)
+        return Background(background_texture, pos, size, shader, geometry)
+
+    @property
+    def pos(self):
+        return self._pos
+
+    @pos.setter
+    def pos(self, value: tuple[float, float]):
+        self._pos = value
+
+    @property
+    def size(self):
+        return self._size
+
+    @size.setter
+    def size(self, value: tuple[int, int]):
+        self._size = value
+        try:
+            self.shader['size'] = value
+        except KeyError:
+            print("Attempting to set uniform 'size' when the shader does not have a uniform with that name.")
+
+    def draw(self, shift=(0, 0)):
+        try:
+            self.shader['pixelTransform'] = self.texture.pixel_transform
+        except KeyError:
+            print("Attempting to set uniform 'pixelTransform' when the shader does not have a uniform with that name.")
+
+        try:
+            self.shader['pos'] = self.pos[0] + shift[0], self.pos[1] + shift[1]
+        except KeyError:
+            print("Attempting to set uniform 'pos' when the shader does not have a uniform with that name.")
+
+        self.texture.use(0)
+
+        self.geometry.render(self.shader)
+
+    @property
+    def blend(self):
+        try:
+            return self.shader['blend']
+        except KeyError:
+            print("Attempting to get uniform 'blend' when the shader "
+                  "does not have a uniform with that name. Defaulting to 1.")
+            return 1
+
+    @blend.setter
+    def blend(self, value):
+        try:
+            self.shader['blend'] = value
+        except KeyError:
+            print("Attempting to set uniform 'blend' when the shader does not have a uniform with that name.")
+
+    def blend_layer(self, other, percent):
+        self.blend = 1 - percent
+        other.blend = percent
+
+
+class BackgroundGroup:
+    """
+    If you have many backgrounds which you would like to draw together and move together this can help.
+
+    The pos of the Background Group is independent of each Background pos.
+    The offset of the BackgroundGroup is the same as each background.
+    """
+
+    def __init__(self):
+        self.backgrounds: list[Background] = []
```

Comment:
It's probably better to keep this as a private member and expose a backgrounds property with proper docstrings.